packages/packages/core/editor-components/src/hooks/use-archive.ts

```diff
@@ -0,0 +1,15 @@
+import { useCallback } from 'react';
+import { __useDispatch as useDispatch } from '@elementor/store';
+
+import { slice } from '../store/store';
+
+export const useArchive = () => {
+	const dispatch = useDispatch();
+
+	return useCallback(
+		( componentId: number ) => {
+			dispatch( slice.actions.archive( componentId ) );
+		},
+		[ dispatch ]
+	);
+};
```

Comment:
i think it can just be an action, not a hook (will also be more helpful for mcp)

```suggestion
export const archiveComponent = ( componentId: number ) => {
	const dispatch = getStore()?.dispatch;

	if ( ! dispatch ) {
		return;
	}

	dispatch( slice.actions.archive( componentId ) );
};
```